src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveCreateTrueMaskTest.template

```diff
@@ -0,0 +1,268 @@
+        //         when '11110' numElem = elements - (elements MOD 3);
+        //         when '11111' numElem = elements;
+        //         otherwise    numElem = 0;
+        //     return numElem;
+
+        private static int DecodePredCount({Op1Type} bitpattern, int esize)
+        {
+            if (bitpattern > {Op1Type}.All)
+            {
+                throw new Exception("Bit patterns must be 5 bits long.");
+            }
+
+            switch (esize) {
+                case 8:
+                case 16:
+                case 32:
+                case 64:
+                    break;
+                default:
+                    throw new Exception("Incorrect esize");
+            }
+
+            int elements = (int)(Sve.Count64BitElements()) * 64 / esize;
+            switch (bitpattern)
+            {
+                case {Op1Type}.LargestPowerOf2:
+                    return FloorPow2(elements);
+                case {Op1Type}.VectorCount1:
+                    return elements >= 1 ? 1 : 0;
+                case {Op1Type}.VectorCount2:
+                    return elements >= 2 ? 2 : 0;
+                case {Op1Type}.VectorCount3:
+                    return elements >= 3 ? 3 : 0;
+                case {Op1Type}.VectorCount4:
+                    return elements >= 4 ? 4 : 0;
+                case {Op1Type}.VectorCount5:
+                    return elements >= 5 ? 5 : 0;
+                case {Op1Type}.VectorCount6:
+                    return elements >= 6 ? 6 : 0;
+                case {Op1Type}.VectorCount7:
+                    return elements >= 7 ? 7 : 0;
+                case {Op1Type}.VectorCount8:
+                    return elements >= 8 ? 8 : 0;
+                case {Op1Type}.VectorCount16:
+                    return elements >= 16 ? 16 : 0;
+                case {Op1Type}.VectorCount32:
+                    return elements >= 32 ? 32 : 0;
+                case {Op1Type}.VectorCount64:
+                    return elements >= 64 ? 64 : 0;
+                case {Op1Type}.VectorCount128:
+                    return elements >= 128 ? 128 : 0;
+                case {Op1Type}.VectorCount256:
+                    return elements >= 256 ? 256 : 0;
+                case {Op1Type}.LargestMultipleOf4:
+                    return elements - (elements % 4);
+                case {Op1Type}.LargestMultipleOf3:
+                    return elements - (elements % 3);
+                case {Op1Type}.All:
+                    return elements;
+                default:
+                    return 0;
+            }
+
+            throw new Exception("Should be unreachable.");
+            return 0;
+        }
+
+        // constant integer VL = CurrentVL;
+        // constant integer PL = VL DIV 8;
+        // constant integer elements = VL DIV esize;
+        // integer count = DecodePredCount(pat, esize);
+        // bits(PL) result;
+        // constant integer psize = esize DIV 8;
+        //
+        // for e = 0 to elements-1
```

Comment:
same here. Please simplify and write the description in text of what we are trying to accomplish.

Reply:
Done.